test/segment-loader.test.js

```diff
@@ -1090,3 +1088,149 @@ QUnit.test('doesn\'t allow more than one monitor buffer timer to be set', functi
 
   QUnit.equal(this.clock.methods.length, timeoutCount, 'timeout count remains the same');
 });
+
+QUnit.module('Segment Percent Buffered Calculations', {
+  beforeEach() {
+    this.env = useFakeEnvironment();
+    this.mse = useFakeMediaSource();
+    this.hasPlayed = true;
+    this.clock = this.env.clock;
+
+    currentTime = 0;
+    loader = new SegmentLoader({
+      currentTime() {
+        return currentTime;
+      },
+      mediaSource: new videojs.MediaSource(),
+      hasPlayed: () => this.hasPlayed
+    });
+  },
+  afterEach() {
+    this.env.restore();
+    this.mse.restore();
+  }
+});
+
+QUnit.test('calculates the percent buffered for segments', function() {
+  let playlist = playlistWithDuration(30);
+
+  loader.mimeType(this.mimeType);
+  loader.playlist(playlist);
+
+  let buffered = videojs.createTimeRanges([[15, 19]]);
+  let percentBuffered = loader.getSegmentBufferedPercent_(playlist, 1, 0, buffered);
```

Comment:
Thought: should we move this utility into the Ranges module? It doesn't have any loader dependency and might keep us from accidentally tying the two together in the future. (Not a blocker for this PR or anything)